spec/features/users/sign_in_spec.rb

```diff
@@ -266,12 +266,22 @@
     end
 
     scenario 'user sees warning before session times out' do
-      expect(page).to have_content(/14 minutes and 5[0-9] seconds/, wait: 5)
-
-      time1 = page.text[/14 minutes and 5[0-9] seconds/]
-      sleep(1.5)
-      time2 = page.text[/14 minutes and 5[0-9] seconds/]
-      expect(time2).to be < time1
+      minutes_and = [
+        t('datetime.dotiw.minutes', count: IdentityConfig.store.session_timeout_in_minutes - 1),
+        t('datetime.dotiw.two_words_connector'),
+      ].join('')
+
+      pattern1 = Regexp.new(minutes_and + t('datetime.dotiw.seconds.other', count: '\d+'))
+      expect(page).to have_content(pattern1, wait: 5)
+      time = page.text[pattern1]
+      seconds = time.split(t('datetime.dotiw.two_words_connector')).last[/\d+/].to_i
+      pattern2 = Regexp.new(
+        minutes_and + t(
+          'datetime.dotiw.seconds.other',
+          count: (((seconds - 10) % 60)...seconds).to_a.join('|'),
```

Comment:
I might remove the modulus math here, since it could get a little hairy when dealing with a situation like `seconds = 5`:

```
(55...5).to_a
# []
```

In practice, because we're waiting for the timer to tick down to the previous minute, it's highly unlikely that we'd ever need to deal with a situation of it ticking into the minute preceding (only if the test took over a minute to run).